linode/manager
# refactor: [M3-8143] - Improve PowerActionsDialog

## Description 📝
This PR helps improve the janky loading state of the `PowerActionsDialog` by removing the flash of content when a dialog or drawer is rendered. The user is now shown helper text in the Reboot Linode dialog above the config select component.

## Changes  🔄
List any change relevant to the reviewer.
- Remove redundant uses of `useLinodeQuery`.
- Pass the `linodeLabel` prop to child component to eliminate the flash of content.
- Replace `default exports` with `named exports`.
- Introduce helper text for the config select component in Reboot Linode dialog.

## Target release date 🗓️
07/22/2024

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![config-before](https://github.com/linode/manager/assets/119514965/18011eb8-fb2e-47a6-a90d-847db819028a) | ![config-after](https://github.com/linode/manager/assets/119514965/39b53422-3415-4f33-bf28-1c0d716532cd) |
| <video src="https://github.com/linode/manager/assets/119514965/517a0f55-6d5d-46f8-84f7-4657b1ca0c17" /> |  <video src="https://github.com/linode/manager/assets/119514965/e608eb09-d5ca-4ca2-81ab-f6a8f65c8299" /> |

## How to test 🧪
### Prerequisites
(How to setup test environment)
- Either:
  - Have at least two linode instances with several configurations saved.
  - Turn on the MSW. 

### Verification steps
(How to verify changes)
- Start with visiting the Linodes landing page.
- Click on the `LinodeActionMenu` and select any of the options.
- Observe that the flash of content has been resolved.
- Verify that the helper text is rendered below the config select in the Reboot Linode dialog.
- Verify that no regressions were introduced with these changes.

## As an Author I have considered 🤔
*Check all that apply*
